app/javascript/packs/project_form_modal.jsx

```diff
@@ -1,44 +1,143 @@
-import React from "react";
+import React from 'react';
+import ReactDOM from "react-dom";
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Form, FormGroup, Label, Input } from 'reactstrap';
 import axios from "axios";
-import ReactDOM from "react-dom"
 
 class ProjectFormModal extends React.Component {
-	state = {
-	  title: "",
-		desc: "",
-		key: "",
-	};
+  constructor(props) {
+    super(props);
+    this.state = {
+      modal: false,
+			data: {
+				title: "",
+				desc: "",
+				key: ""
+			},
+			errors: {}
+    };
+
+    this.toggle = this.toggle.bind(this);
+  }
 
+  toggle() {
+    this.setState(prevState => ({
+      modal: !prevState.modal
+    }));
+  }
 
-	render() {
-	  return (
-			<>
-				<button type="button" className="btn btn-primary" data-toggle="modal" data-target="#createProject">
-				  Create Project
-				</button>
-
-				<div className="modal fade" id="createProject" tabIndex="-1" role="dialog" aria-labelledby="createProjectLabel" aria-hidden="true">
-				  <div className="modal-dialog" role="document">
-				    <div className="modal-content">
-				      <div className="modal-header">
-				        <h5 className="modal-title" id="createProjectLabel">Create Project</h5>
-				        <button type="button" className="close" data-dismiss="modal" aria-label="Close">
-				          <span aria-hidden="true">&times;</span>
-				        </button>
-				      </div>
-				      <div className="modal-body">
-				        ...
-				      </div>
-				      <div className="modal-footer">
-				        <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
-				        <button type="button" className="btn btn-primary">Save changes</button>
-				      </div>
-				    </div>
-				  </div>
-				</div>
-			</>
-	  );
+	setFromValue = (attribute, value) => {
+		this.setState(prev => ({ data: {...prev.data, [attribute]: value} }))
 	}
+
+
+	handleSubmit = e => {
+		if(this.handleValidation()){
+
+	    e.preventDefault();
+
+	    axios
+	      .post(
+	        "project.json",
+	        { project: this.state.data },
+	        {
+	          headers: {
+	            "X-CSRF-Token": document.querySelector('meta[name="csrf-token"]')
+	              .content
+	          }
+	        }
+	      ).catch(e => {
+					if(e.response.data.errors["key"] !== "undefined"){
+						this.setState({errors: {key: "The key must be unique"}});
+					}
+
+				})
+		};
+	};
+
+		handleValidation(){
+        let project = this.state.data;
+        let errors = {};
+        let formIsValid = true;
+
+        //title
+        if(!project["title"]){
+           formIsValid = false;
+           errors["title"] = "Cannot be empty";
+        }
+
+        if(typeof project["title"] !== "undefined"){
```

Comment:
you can use simpler check here `if(project.title)` or just add `else` clause to previous `if`